### PR TITLE
Add Model for Posts

### PR DESCRIPTION
**Pull request by zhpeng811**
A draft for the model on Posts, Comments, and Likes

likely still missing some fields, that can be implemented later when it is required

### CODE REVIEW

File: posts/models.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import uuid
+from django.db import models
+from authors.models import Author
+from django.utils.translation import gettext_lazy as _
+
+
+class Comment(models.Model):
+    id = models.UUIDField(primary_key=True, editable=False, default=uuid.uuid4)
+    author = models.ForeignKey(Author, related_name="commenter", on_delete=models.CASCADE)
```
Comment (c25vdw) [inline posts/models.py:9]: line 9: related_name shows up on the other side indicating how this model is referred to. So let's say an Author objects has some comments, then in current code all his/her comments shows up as `commenter.all()`, which probably doesn't make sense.

how about 
```suggestion
    author = models.ForeignKey(Author, related_name="comments_sent", on_delete=models.CASCADE)
```


Or even, we don't have to have one, and Django should make `comment_set` for us. https://stackoverflow.com/a/2642645 I'm good with both `comments_sent` or `comment_set`, or any name that says "the comments this author sent out to other people"

File: posts/models.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    author = models.ForeignKey(Author, related_name="post", on_delete=models.CASCADE)
+    
+    title = models.CharField(max_length=30) # title should not be super long
+    source = models.URLField(editable=False)
+    origin = models.URLField(editable=False)
+    description = models.CharField(max_length = 50)
+    content_type = models.CharField(max_length=3, choices=ContentType.choices, default=ContentType.PLAIN)
+    content = models.TextField()
+    published = models.DateTimeField()
+    comments = models.ManyToManyField(Comment, related_name="post_comments", blank=True)
```
Comment (c25vdw) [inline posts/models.py:41]: We need to link up Comment many---->one Post, many to many probably doesn't work. 

How about let's not have comments in Post, and instead, have Post as a foreign key in Comment, with this, posts can also have a related name, and that follows the same format as you have with Comment-Author relationship.

So basically, Comment is just a Author-Post relation table (two foreign keys), with extra comment body attached.

File: posts/models.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    def get_public_id(self):
+        return self.url or self.id
+
+    def get_api_type(self):
+        return 'post'
+
+class Like(models.Model):
+    summary = models.CharField(max_length=30)
+    author = models.ForeignKey(Author, related_name = "liker", on_delete=models.CASCADE)
+    post = models.ForeignKey(Post, related_name = "liker", on_delete=models.CASCADE)
```
Comment (c25vdw) [inline posts/models.py:61]: `related_name = "likes"` makes more sense here, line 60 & 61 both. So, an author has all `likes` he sent out; a post has all the `likes` it received from people.

**Review by c25vdw — CHANGES_REQUESTED** [review]
Other than related_name's and the relationship between Post and Comment I don't have complaints.

File: posts/models.py (hunk 1 above)
Comment (zhpeng811) [reply posts/models.py]: removed the related_name so it will be automatically set to `comment_set`

File: posts/models.py (hunk 2 above)
Comment (zhpeng811) [reply posts/models.py]: done, Comment now have the Post foreign key

File: posts/models.py (hunk 3 above)
Comment (zhpeng811) [reply posts/models.py]: updated

**Review by c25vdw — APPROVED** [review]

**Comment by c25vdw** [comment]
LGTM! but probably a `./manage.py makemigrations` & `./manage.py migrate` is needed before it passes checks.

**Pull request merged**